packages/federation/src/managed-federation.ts

```diff
@@ -282,9 +282,155 @@ export async function getStitchedSchemaFromManagedFederation(
   return result;
 }
 
-const DEFAULT_MESSAGE_LOGGER = {
-  ERROR: (message: string) =>
-    console.error('[Managed Federation] Uplink message: [ERROR]', message),
-  WARN: (message: string) => console.warn('[Managed Federation] Uplink message: [WARN]', message),
-  INFO: (message: string) => console.info('[Managed Federation] Uplink message: [INFO]', message),
+const DEFAULT_MESSAGE_LOGGER = (logger: Logger) => ({
+  ERROR: (message: string) => logger.error('[Managed Federation] Uplink message: [ERROR]', message),
+  WARN: (message: string) => logger.warn('[Managed Federation] Uplink message: [WARN]', message),
+  INFO: (message: string) => logger.info('[Managed Federation] Uplink message: [INFO]', message),
+});
+
+export type SupergraphSchemaManagerOptions = Omit<
+  GetStitchedSchemaFromManagedFederationOpts,
+  'lastSeenId'
+> & {
+  maxRetries?: number;
+  minDelaySeconds?: number;
+  retryDelaySeconds?: number;
+  logger?: Logger;
+};
+
+export type Logger = {
+  info: (message: string, ...args: unknown[]) => void;
+  warn: (message: string, ...args: unknown[]) => void;
+  error: (message: string, error?: unknown, ...args: unknown[]) => void;
 };
+
+export class SupergraphSchemaManager {
```

Comment:
Just an idea... This can be an event target and it can emit events such as `schema`, `error`, `message` etc instead of `onSchemaChange`, `onFailure` and `logger`.

Reply:
For type safety, it can even use something like this;
https://github.com/dotansimha/graphql-yoga/blob/main/packages/event-target/typed-event-target/src/index.ts

Reply:
I've tried to do it. I really dislike the API of [dotansimha/graphql-yoga@main/packages/event-target/typed-event-target/src/index.ts](https://github.com/dotansimha/graphql-yoga/blob/main/packages/event-target/typed-event-target/src/index.ts?rgh-link-date=2024-05-31T18%3A50%3A53Z). It's tight to DOM usage which make it much less clear how to use it.

I've tried a implementation based on `EventEmitter` which works pretty great, but it's then tight to Node, which means it would make this work only in Node environment, or I have to make a cross-plateform polyfill for it.